cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/batch/PipelinePhasePreparer.java

```diff
@@ -107,8 +108,20 @@ public PipelinePhasePreparer(PluginContext pluginContext, Metrics metrics, Macro
         submitterPlugin = create(pluginInstantiator, stageSpec);
       }
 
+
       if (submitterPlugin != null) {
-        submitterPlugin.prepareRun();
+        try {
+          submitterPlugin.prepareRun();
+        } catch (TransactionFailureException e) {
+          if (e.getCause() instanceof ValidationException) {
+            ValidationException ve = (ValidationException) e.getCause();
+            throw new IllegalArgumentException(
```

Comment:
is there a need to wrap it in an IllegalArgumentException? It seems better to just throw the ValidationException it as-is, especially because I don't think it gets wrapped in a TransactionFailureException in cloud runs.

Reply:
This is to get the first error message if there is any considering `ValidationException` has a very generic exception message. https://github.com/cdapio/cdap/blob/develop/cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/validation/ValidationException.java#L38

Reply:
Oh I see. I think it should be the responsibility of the app to log all the causes. Perhaps this method can throw the ValidationException and the caller can catch and log all the causes, then rethrow the ValidationException. This would result in a bunch of error messages generated by the app, and then one generic message generated by the platform.